src/generator/resources/Vector.st

```diff
@@ -47,6 +53,91 @@ public class $vector/mutableType()$ extends $vector/baseType()$ {
         this($components:{x| other.get$x.proper$()}; separator = ", "$);
     }
 
+$if(componentType.discrete)$
+    /**
+     $components:{x| * @param $x$ the $x$ component}; separator = "\n"$
+     */
+    public $vector/mutableType()$($components:{x|float $x$}; separator = ", "$) {
+        this($components:{x|
+            DoubleMath.roundToInt($x$, RoundingMode.FLOOR)};
```

Comment:
I suggest moving them to TeraMath, at least for compatibility reasons. It could be a bit faster than DoubleMath, but I doubt that the difference will be large.

We could mark it as `@Deprecated` and refer to the google implementation for a while.